storage/innobase/xtrabackup/test/inc/common.sh

```diff
@@ -790,6 +790,56 @@ function require_qpress()
     fi
 }
 
+function require_tokudb()
+{
+    if ! [ -a $(dirname ${MYSQLD})/../lib/mysql/plugin/ha_tokudb.so ]; then
+        skip_test "Requires TokuDB"
+    fi
+}
+
+##############################################################################
+# Start a server with TokuDB plugins loaded and enabled.
+# Server id is 1, any arguments are passsed to the mysqld.
+##############################################################################
+function start_server_with_tokudb()
+{
+    # Modified piece from ps_tokudb_admin
+    local SCRIPT_PWD=$(dirname ${MYSQLD})
+    local JEMALLOC_LOCATION=""
+
+    # Check location for libjemalloc.so.1
+    printf "Checking location of jemalloc library ...\n"
+    for libjemall in "${SCRIPT_PWD}/lib/mysql" "/usr/lib64" "/usr/lib/x86_64-linux-gnu" "/usr/lib"; do
+      if [ -r "$libjemall/libjemalloc.so.1" ]; then
+	JEMALLOC_LOCATION="$libjemall/libjemalloc.so.1"
+	break
+      fi
+    done
+    if [ -z $JEMALLOC_LOCATION ]; then
+      die "ERROR: Cannot find libjemalloc.so.1 library. Make sure you have libjemalloc1 on debian|ubuntu or jemalloc on centos package installed.\n\n";
```

Comment:
Can we skip the test case if jemalloc isn't found instead of failing it? It doesn't really matter where we set LD_PRELOAD, we can actually do it in `require_tokudb` or we can add `require_jemalloc`.